src/dotty/tools/dotc/core/Names.scala

```diff
@@ -200,44 +203,21 @@ object Names {
   private final val fillFactor = 0.7
 
   /** Memory to store all names sequentially. */
+  @sharable // because it's only mutated in synchronized block of termName
```

Comment:
Keeping track and maintaining such annotation could be a burden. 
Instead of annotating explicitly, could we infer that private fields that are modified in `synchronized(this)` blocks are sharable?